xray/client/xray_viewer.ts

```diff
@@ -98,23 +107,22 @@ export class XRayViewer {
       edge_length_px_for_level *= 2;
       level += 1;
     }
+    this.displayLevel = level;
 
     // We encode the matrix column major.
-    const request = new Request(
-      `${this.prefix}/nodes_for_level?level=${level}&matrix=${matrixToString(
-        matrix
-      )}`,
-      {
-        method: 'GET',
-        credentials: 'same-origin',
-      }
-    );
-
+    this.currentForLevelQueryNumber++;
```

Comment:
Is this basically just a unique ID? If so, I'd suggest naming it `nextRequestId`.

Reply:
see above.